tools/summaries/fetch_place_summaries.py

```diff
@@ -165,26 +169,81 @@ def build_template_summaries(place_dcids: List[str], stat_var_json=str) -> Dict:
   return summaries
 
 
-def main():
-  with open("priority-places.json") as f:
-    priority_places = json.load(f)
-
-  with open("places-countries.json") as f:
-    countries = json.load(f)
-
-  priority_summaries = build_template_summaries(place_dcids=priority_places,
-                                                stat_var_json=_STAT_VAR_JSON)
-
-  country_summaries = build_template_summaries(place_dcids=countries,
-                                               stat_var_json=_STAT_VAR_JSON)
-
-  all_summaries = utils.combine_summaries(
-      [priority_summaries, country_summaries])
-
-  utils.write_summaries_to_file(summaries=all_summaries,
-                                output_file=_OUTPUT_FILENAME)
+def build_template_summaries_for_sitemap(
+    sitemap: str,
+    stat_var_json: str = _STAT_VAR_JSON,
+    batch_size: int = _BATCH_SIZE,
+    output_file: str = _OUTPUT_FILE) -> Dict:
+  """Generate summaries for all places in a sitemap"""
+  start_time = time.time()
+
+  # Extract places to create summaries for from sitemap
+  places = utils.get_places_from_sitemap(sitemap)
+  total_num_places = len(places)
+  logging.info(f'Generating summaries for {total_num_places} places')
+
+  # Get summaries in batched calls
+  batch_num = 0
+  batch_start_time = time.time()
+  batches = utils.batched(places, batch_size)
+  total_num_batches = len(batches)
+  for batch in utils.batched(places, batch_size):
```

Comment:
nit: just iterate through batches instead of re-creating the batches?

Reply:
Good call, updated.